tests/rptest/tests/franz_go_verifiable_test.py

```diff
@@ -118,7 +118,14 @@ class FranzGoVerifiableWithSiTest(FranzGoVerifiableBase):
 
     segment_size = 5 * 1000000
 
-    topics = (TopicSpec(partition_count=100, replication_factor=3), )
+    topics = (TopicSpec(
+        partition_count=10,
+        replication_factor=3,
+        segment_bytes=segment_size,
```

Comment:
I do like specifying the segment bytes during topic creation, but it ends up a little bit duplicative because SISettings is also taking a `log_segment_size` and using  it to set the redpanda default segment size.

If this was the only place using that, then it would be neat to remove the segment size from SISettings, and have One True Place for setting that parameter (the new one in TopicSpec).